packages/fuchsia_remote_debug_protocol/lib/src/common/logging.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:core';
+
+/// Determines the level of logging.
+///
+/// Verbosity is increasing from zero (none) to four (fine). The fifth level
+/// (all) logs everything.
+enum LoggingLevel {
+  /// Logs no logs.
+  none,
+
+  /// Logs severe logs at the most.
```

Comment:
I think these docs should provide some guidance on how these levels should be treated by, e.g. a custom implementation of `LoggingFunction`. Typically "severe" means that an unrecoverable error has been encountered and the process will or should quit. "warning" means that an error happened but the process will attempt to continue and may or may not succeed. The default implementation of `LoggingFunction` should also uphold this contract, (e.g. quit or throw upon "severe" error).

Reply:
Done.  Also implemented `defaultLoggingFunction` to exit in the event of a severe log, and changed the logic so that severe logs are always logged.